lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -84,7 +84,10 @@ def event_action_tuple(event)
       params[:_type] = get_event_type(event) if use_event_type?(nil)
 
       if @pipeline
-        params[:pipeline] = event.sprintf(@pipeline)
+        value = event.sprintf(@pipeline)
+        if value
+          params[:pipeline] = value
+        end
```

Comment:
The expectation is that an empty string results in the parameter not being included. However, in Ruby only `false` and `nil` (the objects, not the strings) are falsey. So an expression with an empty string evalues to true:

```ruby
> value = ""
=> ""
> puts "hi" if value
hi
```

Since `sprintf` always returns a string, we can change this expression to check if it is empty:

```suggestion
        # convention: empty string equates to not using a pipeline
        # this is useful when using a field reference in the pipeline setting, e.g.
        #          pipeline => "%{[metadata][pipeline]}"
        params[:pipeline] = value unless value.empty?
```

Reply:
ha, you can tell I'm used to python since I thought `""` would evaluate to false. thanks for the suggestion.